### ISSUE TITLE
Can't use Mongock without Spring

### ISSUE THREAD

**Issue by mmindenhall** [cloudyrock/mongock#15]
It is not possible to use Mongock without Spring given the current declaration of the class:

```java
public class Mongock implements InitializingBean, Closeable {
// ...
}
```
I think the fix here should be pretty easy...just create a separate subclass that adds in the Spring capabilities:

```java
public class Mongock implements Closeable {
// ...
}

public class SpringMongock extends Mongock implements InitializingBean {
// ...
}

```

**Comment by mmindenhall** [comment]
Hi @dieppa, thanks for fixing #12 and spinning a new release!

To clarify, it may be possible to use Mongock without actually _using_ Spring, but it's impossible to exclude Spring from the project as a dependency because there's no code path that does not touch something from Spring.

I have this in my build.gradle:

```gradle
    compile("com.github.cloudyrock:mongock:1.14") {
        exclude group: 'org.springframework', module: 'spring-beans'
        exclude group: 'org.springframework', module: 'spring-context'
        exclude group: 'org.springframework', module: 'spring-data-mongodb'
    }
```

And when I try to build my project, I get this error:

```
e: Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class com.github.cloudyrock.mongock.Mongock, unresolved supertypes: org.springframework.beans.factory.InitializingBean
```

I think the cleanest way to do this would be to split into separate modules as was done in [mongobee PR#64](https://github.com/mongobee/mongobee/pull/64).

**Comment by dieppa** [comment]
Ahh ok, yeah, that's right.
The solution I am going to provide ti split the project into 2(core and spring), so you would be importing just the core. You can expect that upgrade for tomorrow, does it work for you?

**Comment by dieppa** [comment]
Oh yeah, I didn't see your last comment. That's what I mean't create 2 modules.

**Comment by dieppa** [comment]
Can you wait until tomorrow?

**Comment by dieppa** [comment]
Merged PR #17 . 
Releases artifacts, version 1.15 of mongock-core, mogock-spring, mongock-jongo

<dependency>
  <groupId>com.github.cloudyrock</groupId>
  <artifactId>mongock-core</artifactId>
  <version>1.15</version>
</dependency>


<dependency>
  <groupId>com.github.cloudyrock</groupId>
  <artifactId>mongock-spring</artifactId>
  <version>1.15</version>
</dependency>


<dependency>
  <groupId>com.github.cloudyrock</groupId>
  <artifactId>mongock-jongo</artifactId>
  <version>1.15</version>
</dependency>
